File: 2015/Day1/day1.cpp
#include <iostream>
#include <fstream>
#include <string>

using namespace std;

int main()
{
// Create a text string, which is used to output the text file
string floor_commands;
// Read from the text file
ifstream file("Day1.txt");
// Use a while loop together with the getline() function to read the file line by line
while (getline(file, floor_commands)){}
// Close the file
file.close();

bool basementFound = false;
//Part 1

int floor = 0;

for (int i = 0; i < floor_commands.length(); i++) {
    if (floor_commands[i] == '(') {
        floor += 1;
    }
    else if (floor_commands[i] == ')') {
        floor -= 1;
    }

    if (floor == -1 && !basementFound) {
        cout<< i + 1<< endl;
        basementFound = true;
    }
}
cout<< floor<< endl;
}

File: 2015/Day1/day_1.py
with open("Day1.txt", "r") as file:
    # Read the contents of the file into a variable
    floor_commands = file.read()

    # Print the names
    # print(type(floor_commands))
    # print(floor_commands)

# Part 1
count = 0

for iteration, element in enumerate(floor_commands):
    if element == "(":
        count += 1
    elif element == ")":
        count -= 1

    if count == -1:
        print(iteration + 1)
        # print(count)

    # print(iteration)
    # print(element)

print(count)
